src/repo.rs

```diff
@@ -59,6 +59,42 @@ impl Repository {
         }
     }
 
+    /// Find and open an existing repository, with additional options.
+    ///
+    /// If flags contains REPOSITORY_OPEN_NO_SEARCH, the path must point
+    /// directly to a repository; otherwise, this may point to a subdirectory
+    /// of a repository, and `open_ext` will search up through parent
+    /// directories.
+    ///
+    /// If flags contains REPOSITORY_OPEN_CROSS_FS, the search through parent
+    /// directories will not cross a filesystem boundary (detected when the
+    /// stat st_dev field changes).
+    ///
+    /// If flags contains REPOSITORY_OPEN_BARE, force opening the repository as
+    /// bare even if it isn't, ignoring any working directory, and defer
+    /// loading the repository configuration for performance.
+    ///
+    /// ceiling_dirs specifies a list of paths (separated by the platform path
+    /// list separator) that the search through parent directories will stop
+    /// before entering.  Use the functions in std::env to construct or
+    /// manipulate such a path list.
+    pub fn open_ext<P: AsRef<Path>>(path: P,
+                                    flags: RepositoryOpenFlags,
+                                    ceiling_dirs: Option<&str>)
```

Comment:
I experimented with making `ceiling_dirs` use `Option<T>` where `T: IntoCString`.  However, I found that that created an annoying usability problem for callers: `None` doesn't disambiguate the type inside the `Option`, so callers would have to specify the type of the `None`.

Because of this ambiguity, I don't think IntoCString will work well here for a type that needs to live inside an Option.  Given that, I think it makes sense to stick with accepting a slice, as the current code does.

Any other suggestions to make this interface more usable?